3D IGF Solver

ax3l opened the pull request:
Testing the 3D IGF Poisson Solver.

- [x] add FFTW in CI
- [x] add FFTW in CPU HPC system profiles 
- [x] add runtime option
- [x] add tests
- [x] dependencies
  - [x] build system control https://github.com/ECP-WarpX/WarpX/pull/5051
  - [x] ensure requirements are met: https://github.com/ECP-WarpX/WarpX/pull/4648#discussion_r1595035952 -> https://github.com/ECP-WarpX/WarpX/pull/4945

## Notes

Compiled with
```console
cmake -S . -B build -DImpactX_FFT=ON
```

ax3l commented:
### cfchannel_spacecharge

Increasing the resolution to `72, 72, 64` cells let's the precision pass.

More cleverly, keeping the number of cells but improving the resolution by setting `1.01` (and relying on the IGF open boundaries) passes as well :tada:

ax3l committed:
CI: FFTW

ax3l committed:
Poisson Solver Runtime Control

ax3l committed:
IGF (FFT) Solver: Tests

ax3l committed:
ABLASTR: Update Latest `development`

Update ABLASTR/WarpX to the latest commit in the `development`
branch, b58343a9cf31ad2e2e0c47b2c5f0b15cc122a961.

ax3l committed:
Doc: Perlmutter & Lumi FFTW

ax3l commented:
This is now ready for review & merge :tada:

cemitch99 commented on cmake/dependencies/ABLASTR.cmake:
```diff
@@ -178,7 +178,7 @@ set(ImpactX_openpmd_src ""
     CACHE STRING
     "Repository URI to pull and build ABLASTR from if(ImpactX_ablastr_internal)")
-set(ImpactX_ablastr_branch "24.07"
+set(ImpactX_ablastr_branch "b58343a9cf31ad2e2e0c47b2c5f0b15cc122a961"
```
What is the origin of the string on line 181?

ax3l replied:
That is the hash of the latest commit in WarpX:
- https://github.com/ECP-WarpX/WarpX/commits/development/
- https://github.com/ECP-WarpX/WarpX/commit/b58343a9cf31ad2e2e0c47b2c5f0b15cc122a961

We usually use tags, aka aliases for a commit, but since we are between releases I used a specific commit in `development`.

cemitch99 reviewed (approved):
Looks ready to go.

ax3l commented:
Oops, unrelated to this PR the `pytest` tests of AMReX (`pytest.AMReX` in CTest) are failing.

Update: fixed via https://github.com/AMReX-Codes/pyamrex/pull/341

ax3l committed:
CTest: FFT Tests if Compiled in

Add tests that use the IGF solver only if `ImpactX_FFT` is ON.

ax3l committed:
Update pyAMReX to latest `development`

Commit ff4643869c63d4ee40a87054b901f61eefcb97a3 includes the fix from
https://github.com/AMReX-Codes/pyamrex/pull/341

ax3l committed:
CI: FFTW (CodeQL)
